server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalMappedTerms.java

```diff
@@ -51,7 +52,14 @@ protected InternalMappedTerms(String name, BucketOrder reduceOrder, BucketOrder
      */
     protected InternalMappedTerms(StreamInput in, Bucket.Reader<B> bucketReader) throws IOException {
         super(in);
-        docCountError = in.readZLong();
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) { // todo fix after backport
+            docCountError = in.readOptionalLong();
```

Comment:
Should this stay a zlong?

Reply:
@nik9000 why was it a zlong in the first place? Can errors be negative?